product_category_tax/models/product_template.py

```diff
@@ -34,10 +34,16 @@ def set_tax_from_category(self):
     def create(self, vals_list):
         for vals in vals_list:
             if vals.get("categ_id"):
-                if "taxes_id" not in vals:
-                    categ = self.env["product.category"].browse(vals["categ_id"])
-                    vals["taxes_id"] = [(6, 0, categ.taxes_id.ids)]
-                if "supplier_taxes_id" not in vals:
-                    categ = self.env["product.category"].browse(vals["categ_id"])
-                    vals["supplier_taxes_id"] = [(6, 0, categ.supplier_taxes_id.ids)]
+                existing_taxes = supplier_existing_taxes = set()
+                categ = self.env["product.category"].browse(vals["categ_id"])
+                if "taxes_id" in vals:
+                    existing_taxes = set(vals["taxes_id"][0][2])
+                taxes = existing_taxes | set(categ.sudo().taxes_id.ids)
+                vals["taxes_id"] = [(6, 0, taxes)]
+                if "supplier_taxes_id" in vals:
+                    supplier_existing_taxes = set(vals["supplier_taxes_id"][0][2])
+                supplier_taxes = supplier_existing_taxes | set(
+                    categ.sudo().supplier_taxes_id.ids
+                )
+                vals["supplier_taxes_id"] = [(6, 0, supplier_taxes)]
```

Comment:
as this block of code is pretty much the same as the one used in product_product, maybe we can only keep it in one model and reuse